src/du_parameters/frequency.py

```diff
@@ -0,0 +1,316 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Define Frequency, GSCN and ARFCN classes by overloading arithmetic operators.
+
+Handle arithmetic operations with different types (Decimal, int, float).
+"""
+
+import logging
+from dataclasses import dataclass
+from decimal import Decimal
+
+logger = logging.getLogger(__name__)
+
+
+class Frequency(Decimal):
+    """Represents a frequency with unit conversion methods and arithmetic capabilities."""
+
+    class Units:
+        """Define Units for Frequency class."""
+
+        KHZ = 1000
+        MHZ = 1000 * 1000
+
+    @classmethod
+    def from_khz(cls, value: int) -> "Frequency":
+        """Convert KHz to Frequency."""
+        return cls(Decimal(str(value)) * cls.Units.KHZ)
+
+    @classmethod
+    def from_mhz(cls, value: float) -> "Frequency":
```

Comment:
```suggestion

    @classmethod
    def from_mhz(cls, value: int | str) -> "Frequency":
```

I think we want to avoid floats, we can always pass a floating point number as a string to Decimal and keep the precision. I know you are calling `str()` on the the value in the method and it fixes this issue:

```
>>> Decimal(3.3)
Decimal('3.29999999999999982236431605997495353221893310546875')
>>> Decimal('3.3')
Decimal('3.3')
```

but I think we should avoid floats as much as possible in this module.

Reply:
Done